File: src/data/courses/dsa/arrays-and-hashing/array-manipulation/pascals-triangle/code/cpp/pascals_triangle_sol1.cpp
class Solution {
public:
    // O(n²) — Build each row from previous row
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result;
        for (int i = 0; i < numRows; i++) {
            vector<int> row(i + 1, 1);           // Fill with 1s
            for (int j = 1; j < i; j++)
                row[j] = result[i-1][j-1] + result[i-1][j]; // Sum of two above
            result.push_back(row);
        }
        return result;
    }
};

File: src/data/courses/dsa/arrays-and-hashing/array-manipulation/pascals-triangle/code/java/PascalsTriangleSol1.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<>(Collections.nCopies(i+1, 1));
            for (int j = 1; j < i; j++) row.set(j, res.get(i-1).get(j-1)+res.get(i-1).get(j));
            res.add(row);
        }
        return res;
    }
}